presto-main/src/main/java/com/facebook/presto/operator/aggregation/MaxBy.java

```diff
@@ -123,7 +123,10 @@ public static void combine(Type keyType, MaxOrMinByState state, MaxOrMinByState
             state.setKey(otherState.getKey());
             state.setValue(otherState.getValue());
         }
-        else if (keyType.compareTo(otherState.getKey(), 0, state.getKey(), 0) > 0) {
+        else if (otherState.getKey() == null) {
```

Comment:
I don't think there's a way for this to be NULL in an actual query. Do you have a query that reproduces this?